### PR TITLE
feature: Export PDFs with paged.js (plus refreshed HTML exports)

### LINKED ISSUE #577: Re-enable horizontal_rule nodes in Pub exports when Pandoc 2.8 is released.
[(See this issue in the Pandoc repo for context)](https://github.com/jgm/pandoc/issues/5801)

The Pandoc TeX exporter generates markup for horizontal rules that's apparently not accepted by the latest version of the `xelatex` PDF renderer, causing any PDF export with a horizontal rule to break. For the time being I'm just going to filter our `horizontal_rule` nodes out of PDF exports for Pubs. The issue has [already been fixed](https://github.com/jgm/pandoc/commit/d9db76dcf40d1930ad15317d76fd2c90d9114801) by the Pandoc developers, so once they cut a new release we should update the Pandoc binary that PubPub uses and re-enable horizontal rules.

### LINKED ISSUE #613: Pagedjs
- Update PDF output to use pagedjs
- Fix issues with other outputted formats including Pandoc latex markup.

### LINKED ISSUE #627: EPUB export contains TeX markup
_We use GitHub Issues for tracking active code development. If you have general questions or feedback about PubPub, we invite you to use our [Discourse forum](https://discourse.knowledgefutures.org)._

### Required
---

**What went wrong, step-by-step?**
1. I have a document in PubPub that seems fine. It was initially created by uploading a Word doc
2. I try exporting as an EPUB (after successfully exporting a PDF).
3. When I open the EPUB, there's a lot of TeX-y commands at the top of the first page; similar content appears elsewhere in the book if I go to different sections in the Table of Contents.
4. Probably related: the cover preview contains an error message (see attached screenshot)

The actual content of the document all appears to be there; I don't see anything that's missing (though I haven't checked exhaustively). 

Additionally, running the EPUB through an EPUB validator (such as https://www.ebookit.com/tools/bp/Bo/eBookIt/epub-validator ) brings up a long list of errors.

...

**What URL can we use to see the issue?**

https://dbvisel.pubpub.org/pub/zpggskcf

Choose Download > Epub, then open the downloaded EPUB. I'm opening in Calibre, though I don't think it matters that much.

**What browser/operating system were you using?**

OS X 10.14.6 / Firefox 71.0b9 

### Optional

---

**Have any screenshots? (Attach them by dragging the image here)**

Screen Shot 2019-11-15 at 10 41 42 AM
Screen Shot 2019-11-15 at 10 41 28 AM
Screen Shot 2019-11-15 at 10 45 48 AM
Screen Shot 2019-11-15 at 10 44 16 AM

**Who reported this? (If not you)**

### PR DESCRIPTION
**Pull request by idreyn**
This PR updates our document exporter to use [Paged.js](https://www.pagedmedia.org/paged-js/) rather than a LaTeX-based toolchain to create PDFs. The old pipeline was like this:

```
prosemirror -> html -> pandoc AST -> TeX source -> [xelatex] -> PDF
```

The new pipeline is like this:

```
prosemirror -> html -> [paged.js] -> PDF
```

This avoids a whole class of issues that xelatex has presented to us, which are mostly outside of our control. Moreover, HTML is a comparatively expressive format that will allow us to continue to customize our exports on a per-community and per-Pub basis.

_Screenshots:_
Here is the new PDF title page:
image

_Test plan:_
- I exported several pubs to HTML and PDF and checked that things work as expected, including footnotes and citations, tables, page numbering, and more
- I also checked to make sure that other document formats (which are still rendered by Pandoc) are still exporting as expected.

Resolves #613 
Resolves #627
Resolves #577

### CODE REVIEW

**Comment by isTravis** [comment]
Should this be based against the dev branch?

File: workers/tasks/export/html.js [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+						{bullet}
+						{primaryCollectionTitle}
+					</span>
+				)}
+			</h3>
+			<h1 className="title">{title}</h1>
+			{attributions.length > 0 && (
+				<div className="byline">
+					<h3>
+						{attributions.map((attr, index) => {
```
Comment (isTravis) [inline workers/tasks/export/html.js]: This needs to filter for `isAuthor`. Currently, it is displaying all contributors on the title page.

File: workers/tasks/export/html.js [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+export const createStaticHtml = async (
+	{ prosemirrorDoc, pubMetadata, citations, footnotes },
+	targetPandoc,
+	targetPaged,
+) => {
+	const { title } = pubMetadata;
+
+	const renderableNodes = [
+		filterNonExportableNodes,
```
Comment (isTravis) [inline workers/tasks/export/html.js]: Something here seems to not be catching. I'm getting an `Unknown node type: discussion` error when trying to export [Resisting Reduction](https://jods.duqduq.org/pub/resisting-reduction).

Screen Shot 2019-11-19 at 10 49 19 AM

**Comment by isTravis** [comment]
Do we have styling control over the top line margin content? The alignment is off when there is both a left-hand Community Title and right-hand Pub Title. 

Screen Shot 2019-11-19 at 10 55 29 AM

Further, some pubs seem to not show the Community title in the exported PDF.

Screen Shot 2019-11-19 at 10 57 43 AM

File: workers/tasks/export/html.js (hunk 1 above)
Comment (idreyn) [reply workers/tasks/export/html.js]: 👍

File: workers/tasks/export/html.js (hunk 2 above)
Comment (idreyn) [reply workers/tasks/export/html.js]: Fixed! I think this is happening in prod, too :|

**Comment by idreyn** [comment]
> Do we have styling control over the top line margin content? The alignment is off when there is both a left-hand Community Title and right-hand Pub Title.

image

Looking at the way the polyfill works, I don't see a simple fix. The margin boxes become sibling divs of identical height, with `display: flex; align-items: center` — so there's nothing obvious that would enforce that the text in each box start in the same horizontal line. It strikes me as something that we could fix someday with CSS subgrids.

> Further, some pubs seem to not show the Community title in the exported PDF.

Yup, this was an issue with the way the  `content` CSS property handles concatenation — just fixed it.

**Pull request merged**